cupy/linalg/_decomposition.py

```diff
@@ -228,11 +228,7 @@ def _qr_batched(a, mode):
             return (cupy.empty(batch_shape + (m, 0), out_dtype),
                     cupy.empty(batch_shape + (0, n), out_dtype))
         elif mode == 'complete':
-            if batch_size == 0:
-                q = cupy.empty(batch_shape + (m, m), out_dtype)
-            else:
-                q = cupy.eye(m)
-                q = cupy.stack([q for i in range(batch_size)])
```

Comment:
`dtype=out_dtype` and `.reshape` were missing.

Reply:
Good catch, thanks!